Sources/PubNubChat/Storage/CoreData/CoreDataMigrationManager.swift

```diff
@@ -63,20 +64,45 @@ class DefaultCoreDataMigrationManager: CoreDataMigrationManager {
       return
     }
     
-    for modelVersion in nextModelVersions {
+    let allVersionFiles = modelBundle.paths(
+      forResourcesOfType: "mom",
+      inDirectory: modelURL.lastPathComponent
+    )
+    
+    let sortedManagedObjectModels = allVersionFiles.compactMap() {
+      NSManagedObjectModel(contentsOf: URL(fileURLWithPath: $0))
+    }.sorted() {
+      $0.integerValue < $1.integerValue
+    }
+    
+    let currentStoreMetadata = try NSPersistentStoreCoordinator.metadataForPersistentStore(
+      ofType: NSSQLiteStoreType,
+      at: persistentStoreLocation
+    )
+    
+    let identifiers = (currentStoreMetadata["NSStoreModelVersionIdentifiers"] as? [String] ?? []).first ?? String()
```

Comment:
it doesn't look good, is it possible to NSStoreModelVersionIdentifiers array not exists or be empty?

Reply:
Yeah, it doesn't exist if you select in Xcode the `PubNubChatModel` file. That's why `0` is used in next lines below as a default value which means this is the oldest CoreData model that ever existed.

Next model versions like `PubNubChatModel-V2` and `PubNubChatModel-V3` already had an integer value set in ascending order. You can check it in Xcode under the `Identifier` field so we're able to distinguish models between themselves. This is what we also should set for every new model version because it's difficult or impossible to perform a migration without having it. Anyway, I will describe you guys the problems and possible choices regarding migrations in a separate ADR.

![image](https://user-images.githubusercontent.com/102806147/182360803-696367e3-30c6-4b48-9005-914da02540b8.png)